pipeline/data_store.py

```diff
@@ -136,6 +136,11 @@ def parse_args(self, *args, **kwargs):
         Can initialize based on exposure and section ids,
         or give a specific image id or coadd id.
 
+        If given an Image that is already loaded with related products
+        (SourceList, PSF, etc.) then these will also be added to the
+        datastore's attributes, to be checked against Provenance in the
+        usual way when the relevant getter is called (e.g., get_sources).
```

Comment:
With all the related products being stored in the Image object, I'm less clear on what the data_store does for us.  I know it has the various methods for saving things and adding them to the database, but at this point might it make more sense just to put all of those in Image?

Reply:
I thought about this, too. I think the datastore still does some complicated work for us. 
So here's what the Image can do: it can, if it has upstream_images and a provenance, load the correct products for each of the upstream images. It can also align them. 

The datastore can fetch not only upstream images, it can fetch the image for an exposure, data products for a single image (if they exist), the reference for an image. These all require non-trivial provenance logic. I am not sure it is a good idea to swap it for the Image object. I will have to think more on this. 

I am also going to add some failure modes to the data store, adding to the complexity of pushing all of this into the Image.